src/editor/editor_page.pas

```diff
@@ -566,7 +568,16 @@ procedure TEditorPage.BeforeStatementHandler(Statement: TCustomStatement);
 
   Idx := Statement.LineNo - 1;
   if (Idx >= 0) then
-    FOutputCreator.DoCommand('.' + OutputCreatorNormalizeText(FExecutingLines[Idx]));
+    // parse out the actual command being run
+    // no output for begin; or end;
+    begin
+      S := OutputCreatorNormalizeText(copy(FExecutingLines[Idx],Statement.ColNo,1000));
+      CIdx := pos(';',S);
+      if (CIdx > 0)  then
+        S := copy(S,0,CIdx);
+      if (S <> 'begin;') then
```

Comment:
What happens if we write:
* BeGiN;
* begin ;
I can't remember if the parser is case aware, but something tells me it does not care.

The comparison should always be done in a case-insensitive matter.